actor.go

```diff
@@ -37,7 +43,28 @@ func NewActorStart(name string, v ...interface{}) *ActorStart {
 	}
 }
 
-func init() {
-	Register(Ack{})
-	Register(ActorStart{})
+type makeActorRegistry struct {
+	mu sync.RWMutex
+	r  map[string]MakeActor
+}
+
+func newMakeActorRegistry() *makeActorRegistry {
+	return &makeActorRegistry{
+		r: make(map[string]MakeActor),
+	}
+}
+
+func (r *makeActorRegistry) Get(name string) (m MakeActor, found bool) {
+	r.mu.RLock()
+	defer r.mu.RUnlock()
+	m, found = r.r[name]
+	return
+}
+
+func (r *makeActorRegistry) Set(name string, m MakeActor) (update bool) {
+	r.mu.Lock()
+	defer r.mu.Unlock()
+	_, update = r.r[name]
```

Comment:
How did this even compile? 
```suggestion
```

Reply:
We may need to check travis, maybe we're using an old version of Go for this repo.

Reply:
nm, I didn't notice that you're using a named var in the return.